app/src/keymap.c

```diff
@@ -146,11 +147,20 @@ static inline int set_layer_state(zmk_keymap_layer_id_t layer_id, bool state) {
         return 0;
     }
 
+    // Non-forcing disables should not change a locked active layer
+    if (!locking && !state && ((_zmk_keymap_layer_locks >> layer_id) & 1)) {
```

Comment:
```suggestion
    if (!locking && !state && (BIT(layer_id) & _zmk_keymap_layer_locks)) {
```
Perhaps a bit clearer? (not tested)